src/com/mostc/pftt/util/HostEnvUtil.java

```diff
@@ -713,6 +713,24 @@ private static void downloadFile(FileSystemScenario fs, LocalConsoleManager cm,
 		}
 	}
 
+	/* Adds a rule for the installerFile to bypass the firewall.
+	 * Will not create an additional rule if it already exists.
+	 */
+	private static void addRuleToFirewall(ConsoleManager cm, AHost host, String name, String installerFile) throws IOException, Exception {
+		String rule = name.replace(' ', '_');
+
+		ExecOutput op = host.execOut("netsh advfirewall firewall show rule name=" + rule, AHost.ONE_MINUTE);
+
+		// Check if rule exists for file, if not add it
+		if(op.output.contains("No rules match the specified criteria.")) {
```

Comment:
Can we check the exit status of `netsh` instead of the output? Otherwise we're depending on the locale; e.g. German says "Keine Regeln stimmen mit den angegebenen Kriterien überein."

Reply:
The only exit status that I can find is if it returns 0, then the program exited successfully. So if the rule didn't exit, then it will still return 0. Instead, would it be ok if it checked for the name (`MySql-Server-5.7.25`) itself in the output? 

I updated the code to reflect this. Thoughts?

Reply:
Hmm, got this 
```
C:\WINDOWS\system32>netsh advfirewall firewall show rule name=hellothere

No rules match the specified criteria.


C:\WINDOWS\system32>echo %errorlevel%
1

C:\WINDOWS\system32>
```

Perhaps it's something in the Java implementation, that wipes the exit status?

Thanks.

Reply:
Oh, you could do this perhaps

`netsh advfirewall firewall show rule name=hellothere 2>&1 > nil & echo %errorlevel%`

The question about why there's no exit status is probably for some other PR :)

Thanks.

Reply:
When I run the following command with `host.execOut()`:
`netsh advfirewall firewall show rule name=hellothere 2>&1 > nil & echo %errorlevel%`

I get errorlevel to be 0 from pftt. When testing it manually through CLI, it behaves a little incorrectly. The errorlevel will depend on the last call not the current call. 

If I try to run 2 separate calls with `host.execOut()`
`netsh advfirewall firewall show rule name=hellothere`
`cmd /c echo %errorlevel%`

`errorlevel` ends up being 0 instead of 1 as well for some reason.

Reply:
You're right, the first one doesn't work correctly. With the second one would work on cmd, but seems not to be liked by Java. Ok, was worth while to try :) It would be anyway a bit better that cehcking text, but still not as clean as checking an exit code.

IMO the current solution should be ok for now, so we can move forward on this. But the issue should be noted so the Java mechanism can be fixed later. The exit status issue affects many other places, which all will need to be revisited. @cmb69 would it be a blocker for you?

Thanks.

Reply:
@weltling, no, of course no blocker for me. :) Just adding a TODO marker to the code should be fully sufficient.

Reply:
@lavturo perhaps more reliable what you could do for now is

`powershell -Command "if ($(Get-NetFirewallRule -Name 'hellothere')) {echo 'found'} else { echo 'not found' }" 2>nul`

in that case one could at least control the output, disregarding locale. Play with the property also, there's Name and DisplayName and probably more.

Thanks.

Reply:
Updated with suggestions. Please look over it.

Reply:
@weltling, @lavturo, maybe a better workaround would be to use the `||` operator, and to echo an error message, e.g.
````
netsh advfirewall firewall show rule name=hellothere 2>&1 > nul || echo failed
````
This outputs nothing if `netsh` succeeded, and "failed" otherwise.